sqlitecluster/SQLiteNode.cpp

```diff
@@ -1809,6 +1809,9 @@ void SQLiteNode::_changeState(SQLiteNodeState newState) {
     _localCommitNotifier.notifyThrough(_db.getCommitCount());
 
     if (newState != _state) {
+        // First, we notify all plugins about the state change
+        _server.notifyStateChangeToPlugins(_db, newState);
+
```

Comment:
@iwiznia why did we notify the state change to the plugins before actually changing the state?

Reply:
Hmmmm I don't know. Is it causing issues?

Reply:
Not exactly. I'm working on a PR that will need to check the state, but because we're notifying before changing the state, I'll have to check both the previous state and current state.

So I just wanted to understand why we did it this way. But I think it makes sense that we notify before we start processing things as leader, for example

Reply:
Yeah for onyx updates we definitely want to do it before we start processing things. We could maybe move this to a bit down the method, though not sure if that would help you in any way for your issue